rocket_twin/systems/tank/tank.py

```diff
@@ -15,33 +17,17 @@ class Tank(System):
     ------
     w_out [kg/s]: float,
         mass flow of fuel exiting the tank
-    weight [kg]: float,
-        weight
-    cg [m]: float,
-        center of gravity
+    shape: TopoDS_Solid,
+        pyoccad model of the structure and fuel
+    props: GProp_GProps,
+        model properties
     """
 
     def setup(self):
 
-        # Geometry
-        self.add_inward("weight_s", 1.0, desc="Structure weight", unit="kg")
-        self.add_inward("weight_max", 5.0, desc="Maximum fuel capacity", unit="kg")
-
-        # Inputs
-        self.add_inward("w_in", 0.0, desc="Fuel income rate", unit="kg/s")
-
-        # Flux control
-        self.add_inward("w_out_max", 0.0, desc="Fuel output rate", unit="kg/s")
-        self.add_inward("w_command", 1.0, desc="Fuel output control variable", unit="")
-
-        # Outputs
-        self.add_outward("weight", 1.0, desc="Weight", unit="kg")
-        self.add_outward("cg", 1.0, desc="Center of gravity", unit="m")
-        self.add_outward("w_out", 0.0, desc="Fuel output rate", unit="kg/s")
-
-        # Transient
-        self.add_transient("weight_p", der="w_in - w_out", desc="Propellant weight")
+        self.add_child(
+            TankFuel("fuel"), pulling=["w_out", "w_in", "w_command", "weight_max", "weight_p"]
+        )
+        self.add_child(TankGeom("geom"), pulling=["shape", "props"])
 
-    def compute(self):
-        self.w_out = self.w_out_max * self.w_command
-        self.weight = self.weight_s + self.weight_p
+        self.connect(self.geom.inwards, self.fuel.inwards, ["weight_p"])
```

Comment:
you should not connect 2 inwards. this is equivalent with just pulling "weight_p" from TankGeom('geom')

Reply:
Check